src/main/resources/db/migration/V1_68__create_utilisation_report.sql

```diff
@@ -0,0 +1,307 @@
+    on jp.pool_number = p.pool_no
+    left join juror_mod.appearance a
+    on jp.juror_number = a.juror_number
+    where p.loc_code = a.loc_code
+    and (((jp.status in (2,3,4,10) or (jp.status = 7 and a.attendance_date is not null)) and jp.is_active ='Y')
+          or (jp.status = 8 and a.attendance_date is not null and j.completion_date is not null))
+    and p.loc_code = p_loc_code
+    and ((jp.status = 10 and jp.transfer_date >= p_start_date)
+           or (jp.status <> 10 and (j.completion_date is null or j.completion_date >= p_start_date)))
+    group by jp.juror_number
+    having (least(coalesce(min(a.attendance_date), min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end)),
+               min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end)) <= p_end_date);
+END;
+$function$
+;
+
+
+-- function to return list of appearance for jurors within report window
+CREATE OR REPLACE FUNCTION juror_mod.util_report_appearance_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+  	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	trial character varying,
+	jury_from date,
+	jury_to date,
+	report_date date,
+	within_service boolean,
+	active_trial boolean,
+	attendance boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	a.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+	null::date,
+	null::date,
+	null::character varying,
+	null::date,
+	null::date,
+	a.attendance_date,
+	true,
+	a.sat_on_jury,
+	true,
+	null::boolean
+	from juror_mod.appearance a
+	where a.attendance_date between p_start_date and p_end_date
+	and (a.non_attendance is null or a.non_attendance = false)
+	and (a.no_show is null or a.no_show = false)
+	and a.loc_code= p_loc_code
+	order by a.juror_number, a.attendance_date;
+
+END;
+$function$
+;
+
+
+-- function to return juror participation in a trial within the report window
+CREATE OR REPLACE FUNCTION juror_mod.util_trial_participation_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+   	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	trial character varying,
+	jury_from date,
+	jury_to date,
+	report_date date,
+	within_service boolean,
+	active_trial boolean,
+	attendance boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	jt.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+		null::date,
+	null::date,
+	jt.trial_number,
+	jt.empanelled_date as jury_from,
+	coalesce (jt.return_date, p_end_date) as jury_to,
+	report_dates.report_date,
+	true,
+	case when report_dates.report_date between jt.empanelled_date and coalesce (jt.return_date, p_end_date) then true else false end as active_date,
+	null::boolean,
+	null::boolean
```

Comment:
Given we are not using a "union", do we need to select these columns as null? Can they just be removed from the table/query?